buildpacks/sbt/src/layers/dependency_resolver_home.rs

```diff
@@ -0,0 +1,78 @@
+use crate::SbtBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+use libcnb::Buildpack;
+use std::path::Path;
+
+pub(crate) struct DependencyResolverHomeLayer {
+    pub(crate) available_at_launch: bool,
+    pub(crate) dependency_resolver: DependencyResolver,
+}
+
+pub(crate) enum DependencyResolver {
+    Ivy,
+    Coursier,
+}
+
+impl Layer for DependencyResolverHomeLayer {
+    type Buildpack = SbtBuildpack;
+    type Metadata = GenericMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: self.available_at_launch,
+            cache: true,
+        }
+    }
+
+    fn create(
+        &self,
+        _context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error> {
+        LayerResultBuilder::new(GenericMetadata::default())
+            .env(
+                LayerEnv::new()
+                    .chainable_insert(
+                        get_layer_env_scope(self.available_at_launch),
+                        ModificationBehavior::Delimiter,
+                        "SBT_OPTS",
+                        " ",
+                    )
+                    .chainable_insert(
+                        get_layer_env_scope(self.available_at_launch),
+                        ModificationBehavior::Append,
+                        "SBT_OPTS",
+                        format!(
+                            "-D{}={}",
+                            match self.dependency_resolver {
+                                DependencyResolver::Ivy => "sbt.ivy.home",
+                                DependencyResolver::Coursier => "sbt.coursier.home",
+                            },
+                            layer_path.to_string_lossy()
+                        ),
+                    ),
+            )
+            .build()
+    }
+
+    fn existing_layer_strategy(
+        &self,
+        _context: &BuildContext<Self::Buildpack>,
+        _layer_data: &LayerData<Self::Metadata>,
+    ) -> Result<ExistingLayerStrategy, <Self::Buildpack as Buildpack>::Error> {
+        Ok(ExistingLayerStrategy::Keep)
```

Comment:
(Optional) If the env vars set by this layer ever need to change, then the layer strategy here would need to be updated. Is it worth adding a comment around the `LayerEnv` usages mentioning that? Or alternatively, using the default layer strategy of `Recreate` given that the performance cost of writing out the env files each time is very low?

Reply:
Oh, the env vars will cause `sbt` to write the cache for downloaded dependencies into this layer, this is why it exists. The env vars are just there to configure `sbt` to use this location. Should the property for the cache location ever change (very unlikely) we can always add a version field to the metadata which will invalidate the layer immediately since the metadata cannot be deserialised.